### ISSUE TITLE
Battery level is not included on device status

### ISSUE THREAD

**Issue by strunker** [OpenWonderLabs/SwitchBotAPI#11]
Is that something that can perhaps be added as well? Would be nice to be able to query for that and monitor for when it starts to get low, do not believe the app warns\monitors in its current form.

thanks.

**Comment by DJay-X** [comment]
@SwitchBot-Wonderlabs  Absolutely. Most devices of SwitchBot work with Battery. The battery level of a device should be available via Open API. Looking forward to have this available at a given time in homebridge-switchbot-openapi @donavanbecker

**Comment by nigels0** [comment]
Thirded!

**Comment by MSBIZ** [comment]
Yes, I think it's important to have the information, especially when we have many devices.

**Comment by Nardol** [comment]
I know developers are very busy with new switch-bot products but could this API improvement be implemented? Not sure it will take so much time, it would be only an attribute to add as this information can be get even remotely.

**Comment by alamers** [comment]
Yes, please! :) Also, would be nice to be able to read solar panel connectivity as well. Opens up a lot of opportunities

**Comment by AdyRock** [comment]
Yep, would be very useful. I get it via BLE so don't see why it can't be in the API.

**Comment by strunker** [comment]
Are any of the improvements or enhancements pinned on Git going to be worked? Its been literally years now these issues have been opened for. Thanks.

**Comment by gpuccio87** [comment]
Would be very useful. We could prevent bots from stopping working!

**Comment by ganlbarone** [comment]
> Would be very useful. We could prevent bots from stopping working!

I wouldn't hold your breath. I stopped investing in Switchbot products\code due to lack of API support. It's really not acceptable for a mainstream product, not in 2022.  API first has been a concept for like years now, don't think Switchbot got that memo.

**Comment by Plasmachef** [comment]
Yes, integrate this please!
Thank you very much :-)

**Comment by AdyRock** [comment]
@chenliuyun can you help with this one?

**Comment by slimninja** [comment]
Another +1 on this feature request to return battery percentage via status command.

For example:
```
{
    "statusCode": 100,
    "body": {
        "deviceId": "E2F6032048AB",
        "deviceType": "Bot",
        "currentBattery": 100
    },
    "message": "success"
}
```

**Comment by Francas555** [comment]
Please add possibility to see Smart Lock battery status on Homey. And for other devices.

**Comment by robindelange** [comment]
I'd like to see battery status to, from my Switchbot devices.

**Comment by Minsheng** [comment]
Hi guys, we've managed to implement battery status and firmware version for most of the products.

**Comment by strunker** [comment]
Wow, took over 2 years to complete, but yeah. I will take a look for this later on, this will be a useful addition for sure.

**Comment by slimninja** [comment]
> Hi guys, we've managed to implement battery status and firmware version for most of the products.

Thank you for the hard work! Can you confirm if this is available in the v1.0 API, or only in the V1.1 API? Us HomeAssistant users are still mostly using the V1.0 API (https://community.home-assistant.io/t/switch-bot-api-integration/270550/122)

**Comment by slimninja** [comment]
@Minsheng - It looks like this change has been reverted and is unavailable again via api v1.1. Can you please double check?

When checking status I now only see: {"deviceId": "xxx", "deviceType": "Bot", "hubDeviceId": "xxx"}
